run_tests.py

```diff
@@ -28,91 +110,50 @@ def main():
-        ast_f90 = test.get("ast_f90", False)
-        ast_cpp = test.get("ast_cpp", False)
-        ast_cpp_hip = test.get("ast_cpp_hip", False)
-        ast_openmp = test.get("ast_openmp", False)
-        ast_new = test.get("ast_new", False)
-        asr = test.get("asr", False)
-        asr_preprocess = test.get("asr_preprocess", False)
-        asr_indent = test.get("asr_indent", False)
-        mod_to_asr = test.get("mod_to_asr", False)
-        llvm = test.get("llvm", False)
-        cpp = test.get("cpp", False)
-        c = test.get("c", False)
-        obj = test.get("obj", False)
-        x86 = test.get("x86", False)
-        bin_ = test.get("bin", False)
-        pass_ = test.get("pass", None)
-        pass_with_llvm = test.get("pass_with_llvm", None)
-        if pass_ and pass_ not in ["do_loops", "global_stmts",
-                                   "loop_vectorise", "inline_function_calls"]:
-            raise Exception("Unknown pass: %s" % pass_)
-
-        print(color(style.bold)+"TEST:"+color(style.reset), filename)
-
-        extra_args = "--no-error-banner"
-
-        if ast:
-            run_test("ast", "lpython --show-ast --no-color {infile} -o {outfile}",
-                        filename, update_reference, extra_args)
-
-        if ast_new:
-            run_test("ast_new", "lpython --show-ast --new-parser --no-color {infile} -o {outfile}",
-                        filename, update_reference, extra_args)
-
-        if asr:
-            run_test("asr", "lpython --show-asr --no-color {infile} -o {outfile}",
-                    filename, update_reference, extra_args)
-
-        if pass_ is not None:
-            cmd = "lpython --pass=" + pass_ + " --show-asr --no-color {infile} -o {outfile}"
-            run_test("pass_{}".format(pass_), cmd,
-                     filename, update_reference, extra_args)
-            if pass_with_llvm:
-                cmd = "lpython --pass=" + pass_ + " --show-llvm --no-color {infile} -o {outfile}"
-                run_test("pass_llvm_{}".format(pass_), cmd,
-                        filename, update_reference, extra_args)
-
-        if llvm:
-            if no_llvm:
-                print("    * llvm   SKIPPED as requested")
-            else:
-                run_test("llvm", "lpython --no-color --show-llvm {infile} -o {outfile}",
-                        filename, update_reference, extra_args)
-
-        if cpp:
-            run_test("cpp", "lpython --no-color --show-cpp {infile}",
-                    filename, update_reference, extra_args)
-
-        if c:
-            run_test("c", "lpython --no-color --show-c {infile}",
-                    filename, update_reference, extra_args)
-
-        if tokens:
-            run_test("tokens", "lpython --no-color --show-tokens {infile} -o {outfile}",
-                    filename, update_reference, extra_args)
-
-        print()
-
+        + os.pathsep + os.environ["PATH"]
+    test_data = toml.load(open("tests/tests.toml"))
```

Comment:
A minor suggestion: we can use the absolute path to avoid path errors if we are running that from some other directory. But that's fine, I can fix as a follow up.

Reply:
I merged all PRs. This code now lives in `src/libasr/compiler_tester` and it is shared with LFortran. So you can go ahead and modify /  polish it more.